src/nplinker/pairedomics/podp_antismash_downloader.py

```diff
@@ -28,41 +34,86 @@
+        self.resolved_refseq_id = None if resolved_refseq_id == 'None' else resolved_refseq_id
+        if ast.literal_eval(attempted):
+            self.attempted = True
+        else:
+            self.attempted = False
         self.filename = filename
 
     @classmethod
-    def from_csv(cls, original_id, resolved_id, attempted, filename):
-        return cls(original_id, resolved_id, attempted, filename)
+    def from_csv(cls, original_id, resolved_refseq_id, attempted, filename):
+        return cls(original_id, resolved_refseq_id, attempted, filename)
 
     def to_csv(self):
         return ','.join([
             str(self.original_id),
-            str(self.resolved_id),
+            str(self.resolved_refseq_id),
             str(self.attempted), self.filename
         ])
 
 
-def _get_best_available_genome_id(genome_id_data):
+def _get_genome_status_log(
+        genome_status_file: str | PathLike) -> Dict[str, GenomeStatus]:
+    """Read genome_status.txt in a dict if it exists, otherwise create
+    the dictionary. 
+
+    Args:
+        genome_status_file(str | PathLike): it records genome IDs and local filenames to avoid having to repeat HTTP requests
+        each time the app is loaded (this can take a lot of time if there are dozens of genomes).
+
+    Returns:
+        dict: log for information about the read genome IDs.
+        """
+
+    genome_status = {}
+
+    # 'genome_status.txt' is read, then in the for loop over the genome records it gets updated,
+    # and finally it is saved again in 'genome_status.txt' which is overwritten
+    if os.path.exists(genome_status_file):
+        with open(genome_status_file) as f:
+            for line in csv.reader(f):
+                asobj = GenomeStatus.from_csv(*line)
+                genome_status[asobj.original_id] = asobj
+
+    return genome_status
+
+
+def _get_best_available_genome_id(
+        genome_id_data: Dict[str, str]) -> str | None:
+    """Get the best available ID from genome_id_data dict.
+
+    Args:
+        genome_id_data(dict): dictionary containing information
+        for each genome record present.
+
+    Returns:
+        str | None: ID for the genome, if present, otherwise None. 
+
+        """
     if 'RefSeq_accession' in genome_id_data:
         return genome_id_data['RefSeq_accession']
-    elif 'GenBank_accession' in genome_id_data:
+    if 'GenBank_accession' in genome_id_data:
         return genome_id_data['GenBank_accession']
-    elif 'JGI_Genome_ID' in genome_id_data:
+    if 'JGI_Genome_ID' in genome_id_data:
         return genome_id_data['JGI_Genome_ID']
 
     logger.warning(
-        'No known genome ID field in genome data: {}'.format(genome_id_data))
+        f'No known genome ID field in genome data: {genome_id_data}')
     return None
 
 
-def _ncbi_genbank_search(genbank_id, retry_time=5.0):
+def _ncbi_genbank_search(
+        genbank_id: str,
+        retry_time: float = 5.0) -> Tag | NavigableString | None:
```

Comment:
```suggestion
def _ncbi_genbank_search(
        genbank_id: str,
        retry_times: int = 3) -> Tag | NavigableString | None:
```
The code in this method is repeated due to the "retry".  I'd suggest changing the `retry_time` to `retry_times` and use a `while loop` to control the retry times and to reduce the repeated code, the logics look like below:
```
retry = 1
while retry <= retry_times:
    do_the_http_request_and_parse
    if xxx:
       return dl_element
    retry = retry + 1
    time.sleep(5)

log_failure
return None

```